astropy/utils/compat/misc.py

```diff
@@ -61,15 +78,3 @@ def override__dir__wrapper(self):
         return sorted(members)
 
     return override__dir__wrapper
-
-
-def namedtuple_asdict(namedtuple):
```

Comment:
I felt comfortable doing that because in https://github.com/astropy/astropy/blob/main/astropy/utils/compat/__init__.py
it says

```
The content of this module is solely for internal use of ``astropy``
and subject to changes without deprecations. Do not use it in external
packages or code.
```

They also weren't used, while ``override__dir__`` saw a fair amount of use around astropy, so I didn't feel comfortable removing it outright.